parsing/parsetree.mli

```diff
@@ -1165,7 +1166,8 @@ and module_expr_desc =
       (** [functor(X : MT1) -> ME] *)
   | Pmod_apply of module_expr * module_expr  (** [ME1(ME2)] *)
   | Pmod_apply_unit of module_expr (** [ME1()] *)
-  | Pmod_constraint of module_expr * module_type  (** [(ME : MT)] *)
+  | Pmod_constraint of module_expr * module_type option * modes
+      (** [(ME : MT @@ modes)] *)
```

Comment:
and `(ME @ modes)`, presumably

Reply:
Good catch -- yes, we might as well list the 3 possible constructs